consensus/src/marshal/store.rs

```diff
@@ -0,0 +1,321 @@
+//! Interface for a store of finalized blocks, used by [Actor](super::Actor).
+
+use crate::{
+    simplex::{signing_scheme::Scheme, types::Finalization},
+    Block,
+};
+use commonware_cryptography::{Committable, Digest};
+use commonware_runtime::{Clock, Metrics, Storage};
+use commonware_storage::{
+    archive::{self, immutable, prunable, Archive, Identifier},
+    translator::Translator,
+};
+use std::{error::Error, future::Future};
+
+/// Durable store for [Finalizations](Finalization) keyed by height and commitment.
+pub trait FinalizationStore: Send + Sync + 'static {
+    /// The type of commitment used by the application's [Blocks](Block).
+    type BlockCommitment: Digest;
```

Comment:
The `Commitment` here should be the same for consensus/application. The digest, however, may differ?

Reply:
Left a comment above on this @ https://github.com/commonwarexyz/monorepo/pull/2293#discussion_r2550423532

Reply:
Renamed this associated type to `BlockDigest`, hopefully that's a bit more clear 🙏